packages/next/client/index.js

```diff
@@ -218,6 +237,12 @@ export default async ({ webpackHMR: passedWebpackHMR } = {}) => {
   const renderCtx = { App, Component, props, err: initialErr }
 
   if (process.env.NODE_ENV === 'production') {
+    // kick off static data request now so it's in the cache
+    // when we re-render post-hydration
+    if (data.isFallback) {
+      router._getStaticData(asPath).catch(() => {})
```

Comment:
This might cause 2 data fetches if it doesn't resolve in time. Right?

Reply:
Instead of this, the fallback HTML should include a `<link rel=preload>` for the data (what this effectively does).

Reply:
Updated to use `preload`, good catch 👍